larq_compute_engine/tflite/kernels/bconv2d.cc

```diff
@@ -121,85 +122,19 @@ TfLiteStatus Prepare(TfLiteContext* context, TfLiteNode* node) {
-                       "valid or one-padding.");
-    return kTfLiteError;
-  }
-
-  if (conv_params->channels_in == 0) {
-    // We don't expect this branch to ever be taken because the `channels_in`
-    // attribute was added to the converter at the same time that support for
-    // bitpacked activations was added, but just in case we don't have a value
-    // we should throw here.
-    TF_LITE_KERNEL_LOG(context,
-                       "Cannot read bitpacked input unless the `channels_in` "
-                       "attribute is set in the converter.");
-    return kTfLiteError;
-  }
-
-  if (write_bitpacked_output) {
-    TF_LITE_ENSURE_EQ(context, thresholds->type, kTfLiteInt32);
-  } else {
-    // For 8-bit quantized networks, we support both int8 and float32
-    // post_activation_ values
-    TF_LITE_ENSURE(context,
-                   post_activation_multiplier->type == kTfLiteFloat32 ||
-                       post_activation_multiplier->type == kTfLiteInt8);
-    TF_LITE_ENSURE(context, post_activation_bias->type == kTfLiteFloat32 ||
-                                post_activation_bias->type == kTfLiteInt8);
-
-    // In case that our network is not 8-bit quantized, we do require float
-    // values
-    if (output->type == kTfLiteFloat32) {
-      TF_LITE_ENSURE_EQ(context, post_activation_multiplier->type,
-                        kTfLiteFloat32);
-      TF_LITE_ENSURE_EQ(context, post_activation_bias->type, kTfLiteFloat32);
-    }
-  }
-
-  if (output->type == kTfLiteInt8 &&
-      conv_params->padding_type == kTfLitePaddingSame &&
-      conv_params->pad_value != 1) {
-    TF_LITE_KERNEL_LOG(
-        context,
-        "8-bit quantization is only supported with valid or one-padding");
-    return kTfLiteError;
-  }
-
-  TF_LITE_ENSURE(context, output->type == kTfLiteInt32 ||
-                              output->type == kTfLiteInt8 ||
-                              output->type == kTfLiteFloat32);
+  TF_LITE_ENSURE_EQ(context, input->type, kTfLiteInt32);
+  TF_LITE_ENSURE_EQ(context, filter->type, kTfLiteInt32);
+  TF_LITE_ENSURE_MSG(context,
+                     output->type == kTfLiteInt32 ||
+                         output->type == kTfLiteInt8 ||
+                         output->type == kTfLiteFloat32,
+                     "Supported output types are int8, int32, and float32.");
 
   // Read the filter dimensions
   conv_params->channels_out = SizeOfDimension(filter, 0);
   conv_params->filter_height = SizeOfDimension(filter, 1);
   conv_params->filter_width = SizeOfDimension(filter, 2);
 
-  TF_LITE_ENSURE_EQ(context, filter->type, kTfLiteInt32);
-
-  if (write_bitpacked_output) {
-    TF_LITE_ENSURE_EQ(context, SizeOfDimension(thresholds, 0),
-                      conv_params->channels_out);
-  } else {
-    TF_LITE_ENSURE_EQ(context, SizeOfDimension(post_activation_multiplier, 0),
-                      conv_params->channels_out);
-    TF_LITE_ENSURE_EQ(context, SizeOfDimension(post_activation_bias, 0),
```

Comment:
All of these asserts and checks have essentially been moved around and consolidated.